exporter/splunkhecexporter/client.go

```diff
@@ -299,16 +312,24 @@ func (c *client) fillMetricsBufferMultiMetrics(events []*splunk.Event, buf buffe
 	var permanentErrors []error
 	jsonStream := jsonStreamPool.Get().(*jsoniter.Stream)
 	defer jsonStreamPool.Put(jsonStream)
+	jsonStream.Reset(buf)
```

Comment:
Why do we reset here? shouldn't it be in 320? Otherwise we write the buffer even if when the permanent error is reported `uint(jsonStream.Buffered()) > c.config.MaxEventSize`

Reply:
Good point. Fixing.

Reply:
Ah I see it's subtle - it only is going to be a problem if we have a permanent error on flushing - in that case we need to reset.